src/widgets/ActivityDetails/ui/BackNavigateButton.tsx

```diff
@@ -0,0 +1,21 @@
+import { ArrowLeft } from "react-bootstrap-icons"
+import { useNavigate } from "react-router-dom"
+
+import { BasicButton } from "~/shared/ui"
+import { useCustomTranslation } from "~/shared/utils"
+
+export const BackNavigateButton = () => {
+  const { t } = useCustomTranslation()
+  const navigate = useNavigate()
+
+  const onBackButtonClick = () => {
+    navigate(-1)
```

Comment:
Please add a guard in case when there is no way previous page

Reply:
And if there is no previous page make sure to provide a "falback" route

Reply:
Done, fallback no need here